include/xtensor/xmath.hpp

```diff
@@ -2402,6 +2402,82 @@ XTENSOR_INT_SPECIALIZATION_IMPL(FUNC_NAME, RETURN_VAL, unsigned long long);
 
         return eval(sum(trap, {saxis}));
     }
+
+    /**
+     * @ingroup basic_functions
+     * @brief Returns the one-dimensional piecewise linear interpolant to a function with given discrete data points (xp, fp), evaluated at x.
+     *
+     * @param The x-coordinates at which to evaluate the interpolated values (sorted).
+     * @param The x-coordinates of the data points (sorted).
+     * @param The y-coordinates of the data points, same length as xp.
+     * @param Value to return for x < xp[0].
+     * @param Value to return for x > xp[-1]
+     * @return an one-dimensional xarray, same length as x.
+     */
+    template<class E>
+    inline E interp(const E &x, const E &xp, const E &fp, double left, double right)
+    {
+        // basic checks
```

Comment:
Allowing for different parameter types for `x`, `xp`, `fp` would be better IMO.

Then , instead of using `std::size_t`, I would recommend using `common_size_type_t<E1,  E2, E3>`.

Reply:
It could be 

```
E3 f = E3::from_shape(x.shape());
```

but that might (in hindsight) limiting if the input are for example `int`s

Reply:
Also I'm not sure which type-checks to do.

Reply:
This line surely must be improved, as this usage does not work:

```cpp
inline xt::xtensor<int,2> interp(const xt::xtensor<int,2> &data , size_t N)
{
  xt::xtensor<int,2> out = xt::empty<int>({N,data.shape()[1]});

  xt::xtensor<double,1> x  = xt::linspace(0.0, 1.0, data.shape()[0]);
  xt::xtensor<double,1> xi = xt::linspace(0.0, 1.0, N);

  for ( size_t j = 0 ; j < data.shape()[1] ; j++ )
  {
    auto c  = xt::view(data, xt::all(), j);
    auto ci = xt::view(out , xt::all(), j);

    ci = xt::interp(xi, x, c);
  }

  return out;
}
```

Compiler error:

```bash
/usr/local/include/xtensor/xmath.hpp:2438:20: error: no member named 'from_shape' in 'xt::xview<const
      xt::xtensor_container<xt::uvector<int, std::__1::allocator<int> >, 2, xt::layout_type::row_major,
      xt::xtensor_expression_tag> &, xt::xall<unsigned long>, unsigned long>'
        E3 f = E3::from_shape(x.shape());
               ~~~~^
```